File: backend/models/kermesse.go
package models

import "time"

type Kermesse struct {
	ID            uint      `gorm:"primaryKey" json:"id"`
	Name          string    `gorm:"size:255;not null" json:"name"`
	Description   string    `json:"description"`
	Location      string    `json:"location"`
	StartDate     time.Time `json:"start_date"`
	EndDate       time.Time `json:"end_date"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
	Organisateurs []User    `gorm:"many2many:kermesse_organisateurs;constraint:OnDelete:CASCADE;" json:"organisateurs"` // Relation many-to-many avec les organisateurs
	Participants  []User    `gorm:"many2many:kermesse_participants;constraint:OnDelete:CASCADE;" json:"participants"`   // Relation many-to-many avec les participants
	Stands        []Stand   `gorm:"constraint:OnDelete:CASCADE;" json:"stands"`                                         // Association one-to-many avec les stands
}

File: frontend/lib/models/kermesse.dart
class Kermesse {
  final int id;
  final String name;
  final String description;
  final String location;
  final String startDate;
  final String endDate;

  Kermesse({
    required this.id,
    required this.name,
    required this.description,
    required this.location,
    required this.startDate,
    required this.endDate,
  });

  factory Kermesse.fromJson(Map<String, dynamic> json) {
    return Kermesse(
      id: json['id'],
      name: json['name'],
      description: json['description'] ?? "",
      location: json['location'],
      startDate: json['start_date'],
      endDate: json['end_date'],
    );
  }
}
